src/main/java/jcifs/smb1/netbios/NbtAddress.java

```diff
@@ -0,0 +1,880 @@
+                }
+                IP = ( IP << 8 ) + b;
+                hitDots++;
+            }
+            if( hitDots != 4 || host.endsWith( "." )) {
+                return (NbtAddress)doNameQuery( new Name( host, type, scope ), svr );
+            }
+            return new NbtAddress( UNKNOWN_NAME, IP, false, B_NODE );
+        }
+    }
+
+    public static NbtAddress[] getAllByName( String host,
+                                        int type,
+                                        String scope,
+                                        InetAddress svr )
+                                        throws UnknownHostException {
+        return CLIENT.getAllByName( new Name( host, type, scope ), svr );
+    }
+
+/**
+ * Retrieve all addresses of a host by it's address. NetBIOS hosts can
+ * have many names for a given IP address. The name and IP address make the
+ * NetBIOS address. This provides a way to retrieve the other names for a
+ * host with the same IP address.
+ *
+ * @param host hostname to lookup all addresses for
+ * @throws java.net.UnknownHostException if there is an error resolving the name
+ */
+
+
+    public static NbtAddress[] getAllByAddress( String host )
+                                                throws UnknownHostException {
+        return getAllByAddress( getByName( host, 0x00, null ));
+    }
+
+
+/**
+ * Retrieve all addresses of a host by it's address. NetBIOS hosts can
+ * have many names for a given IP address. The name and IP address make
+ * the NetBIOS address. This provides a way to retrieve the other names
+ * for a host with the same IP address.  See {@link #getByName}
+ * for a description of <code>type</code>
+ * and <code>scope</code>.
+ *
+ * @param host hostname to lookup all addresses for
+ * @param type the hexcode of the name
+ * @param scope the scope of the name
+ * @throws java.net.UnknownHostException if there is an error resolving the name
+ */
+
+
+    public static NbtAddress[] getAllByAddress( String host,
+                                        int type,
+                                        String scope )
+                                        throws UnknownHostException {
+        return getAllByAddress( getByName( host, type, scope ));
+    }
+
+
+/**
+ * Retrieve all addresses of a host by it's address. NetBIOS hosts can
+ * have many names for a given IP address. The name and IP address make the
+ * NetBIOS address. This provides a way to retrieve the other names for a
+ * host with the same IP address.
+ *
+ * @param addr the address to query
+ * @throws UnknownHostException if address cannot be resolved
+ */
+
+    public static NbtAddress[] getAllByAddress( NbtAddress addr )
+                                                throws UnknownHostException {
+        try {
+            NbtAddress[] addrs = CLIENT.getNodeStatus( addr );
+            cacheAddressArray( addrs );
+            return addrs;
+        } catch( UnknownHostException uhe ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)